src/test/java/org/broadinstitute/hellbender/utils/config/ConfigUnitTest.java

```
@@ -0,0 +1,650 @@
+
+        // Assert that our config property is not set:
+        Assert.assertEquals( ConfigFactory.getProperty(BasicTestConfigWithClassPathOverridesAndVariableFile.CONFIG_FILE_VARIABLE_FILE_NAME), null);
+
+        // Set our file location here:
+        ConfigFactory.setProperty(BasicTestConfigWithClassPathOverridesAndVariableFile.CONFIG_FILE_VARIABLE_FILE_NAME, outputDir.getAbsolutePath() + File.separator + overrideFilename);
+
+        // Test with the class that overrides on the class path:
+        final BasicTestConfigWithClassPathOverridesAndVariableFile basicTestConfigWithClassPathOverridesAndVariableFile =
+                ConfigFactory.create(BasicTestConfigWithClassPathOverridesAndVariableFile.class);
+
+        // List properties for inspection:
+        listAndStoreConfigToStdOut(basicTestConfigWithClassPathOverridesAndVariableFile);
+
+        Assert.assertEquals(basicTestConfigWithClassPathOverridesAndVariableFile.booleanDefFalse(), true);
+        Assert.assertEquals(basicTestConfigWithClassPathOverridesAndVariableFile.booleanDefTrue(), false);
+        Assert.assertEquals(basicTestConfigWithClassPathOverridesAndVariableFile.intDef207(), 999);
+        Assert.assertEquals(basicTestConfigWithClassPathOverridesAndVariableFile.listOfStringTest(), new ArrayList<>(Arrays.asList("string4", "string3", "string2", "string1")));
+        Assert.assertEquals(basicTestConfigWithClassPathOverridesAndVariableFile.customBoolean().booleanValue(), false);
+
+        // Reset the config factory:
+        ConfigFactory.clearProperty(BasicTestConfigWithClassPathOverridesAndVariableFile.CONFIG_FILE_VARIABLE_FILE_NAME);
+        Assert.assertEquals(ConfigFactory.getProperty(BasicTestConfigWithClassPathOverridesAndVariableFile.CONFIG_FILE_VARIABLE_FILE_NAME), null);
+    }
+
+    @Test
+    void testOwnerConfigurationWithClassPathOverridesAndVariableFileInput_ParameterizedConfigFileDoesNotExist() throws IOException {
```

A non-existent config file is a silent error (nothing is thrown)?